pkg/controller/plan/adapter/ovirt/builder.go

```diff
@@ -120,10 +123,58 @@ type Builder struct {
 
 // Create DataVolume certificate configmap.
 func (r *Builder) ConfigMap(_ ref.Ref, in *core.Secret, object *core.ConfigMap) (err error) {
-	object.BinaryData["ca.pem"] = in.Data["cacert"]
+	// For CNV 4.21+, ConfigMap is only needed in secure mode (when insecureSkipVerify is not used).
+	// For CNV < 4.21, ConfigMap is required even in insecure mode as a fallback,
+	// since the InsecureSkipVerify field is not supported.
+	if cacert, exists := in.Data["cacert"]; exists && len(cacert) > 0 {
+		object.BinaryData["ca.pem"] = cacert
+	} else {
+		// If no CA cert provided, try to fetch it from the oVirt engine.
+		// This is needed for CNV < 4.21 when insecure mode is enabled but
+		// InsecureSkipVerify field is not supported.
+		cacert, err = r.fetchOVirtCACert()
+		if err != nil {
+			r.Log.Error(err, "Failed to fetch CA certificate from oVirt engine")
+			// Don't return error - let migration proceed and fail with clear error if cert is actually needed
+			err = nil
+		} else if len(cacert) > 0 {
+			object.BinaryData["ca.pem"] = cacert
+		}
+	}
 	return
 }
 
+// Fetches the CA certificate from the oVirt engine URL.
+// This is needed for CNV < 4.21 in insecure mode as a fallback when InsecureSkipVerify field is not supported.
+func (r *Builder) fetchOVirtCACert() (cert []byte, err error) {
```

Comment:
we shouldn't fetch it for them, the users should provide valid certificate for us